### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by yelnikov-andrii**
[DEMO LINK](https://yelnikov-andrii.github.io/react_goods-selector/)

### CODE REVIEW

**Review by anastasiia-nikita — CHANGES_REQUESTED** [review]
Work good!
But in task said: `The clear button should not be visible if there is no selected good`, now this button visible all time.
Try to connect some css framework in next task :)

File: src/App.tsx [hunk 1]
```diff
@@ -14,11 +15,106 @@ const goodsFromServer: string[] = [
...
-  </div>
-);
+type State = {
+  status: string;
+  selected: string;
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    // eslint-disable-next-line react/no-unused-state
```
Comment (Denchepornyuk) [inline src/App.tsx:25]: I believe you can remove this comment

File: src/App.tsx [hunk 2]
```diff
@@ -14,11 +15,106 @@ const goodsFromServer: string[] = [
...
+    selected: 'selected',
+    // eslint-disable-next-line react/no-unused-state
+  };
+
+  selectedGoods = ['Jam'];
+
+  componentDidMount() {
+    // eslint-disable-next-line react/no-access-state-in-setstate
+    this.setState({ status: `${this.selectedGoods} is selected` });
+  }
```
Comment (Denchepornyuk) [inline src/App.tsx:36]: you can remove componentDidMount and add this message to state `${this.selectedGoods} is selected`

File: src/App.tsx [hunk 3]
```diff
@@ -14,11 +15,106 @@ const goodsFromServer: string[] = [
...
+
+class App extends React.Component<{}, State> {
+  state = {
+    // eslint-disable-next-line react/no-unused-state
+    status: 'No goods selected',
+    selected: 'selected',
+    // eslint-disable-next-line react/no-unused-state
+  };
+
+  selectedGoods = ['Jam'];
```
Comment (Denchepornyuk) [inline src/App.tsx:31]: this variable should be in the state

File: src/App.tsx [hunk 4]
```diff
@@ -14,11 +15,106 @@ const goodsFromServer: string[] = [
...
+
+  render() {
+    return (
+      <div className="App">
+        <ul>
+          {
+            // eslint-disable-next-line no-return-assign
+            goodsFromServer.map((good, index) => (
+              <>
+                <li
```
Comment (Denchepornyuk) [inline src/App.tsx:76]: do you need React.Fragment here?

File: src/App.tsx [hunk 5]
```diff
@@ -14,11 +15,106 @@ const goodsFromServer: string[] = [
...
+            ))
+          }
+        </ul>
+        <div>
+          {this.state.status !== 'No goods selected' && (
+            <button
+              type="button"
+              onClick={() => {
+                this.clear();
+              }}
```
Comment (Denchepornyuk) [inline src/App.tsx:108]:
```suggestion
              onClick={this.clear}
```

**Review by Denchepornyuk — CHANGES_REQUESTED** [review]

**Pull request closed without merging**